File: week6/day3/weather.py
import requests
import pprint

api_key = "e82251deb1644f092281f508884933dd" # super secret. don't share!

city_search = input("For what city do you want to know the weather? ")
print()

def make_request(url):
    response = requests.get(url)

    if response.status_code == 200:
        responseJson = response.json()
        return responseJson
    else:
        print("something went wrong")


get_geocode_url = f"http://api.openweathermap.org/geo/1.0/direct?q={city_search}&limit=5&appid={api_key}"

geocode_info = make_request(get_geocode_url)

print(f"Which {city_search}? (choose number)")
for i, res in enumerate(geocode_info):
    print(f"{i + 1}. {res['name']}, {res['state']}, {res['country']}")

user_selection = int(input())
city_selection = geocode_info[user_selection - 1]

lat, lon = city_selection["lat"], city_selection["lon"]

current_weather_data = f"https://api.openweathermap.org/data/2.5/weather?lat={lat}&lon={lon}&appid={api_key}&units=metric"

get_weather_info = make_request(current_weather_data)

pprint.pprint(get_weather_info)

File: week6/day3/weather.js
function fetchApi(url) {
  return fetch(url).then((res) => res.json());
}

const apiKey = 'e82251deb1644f092281f508884933dd'; // super secret.

const city = prompt('What city do want to the weather for?');
// const city = 'London';

const getGeocodeUrl = `http://api.openweathermap.org/geo/1.0/direct?q=${city}&limit=5&appid=${apiKey}`;

function getGeocodes() {
  return fetchApi(getGeocodeUrl);
}

const geocodes = getGeocodes();

geocodes.then((geocode) => {
  // console.log(geocode);
  let locString = '';
  for (let i = 0; i < geocode.length; i++) {
    locString += `${i + 1}. ${geocode[i].name}, ${geocode[i].state}, ${geocode[i].country} \n`;
  }

  const userSelection = prompt(`Make a selection:
    
    ${locString}
    `);

  const locationSelection = geocode[userSelection - 1];

  const { lat, lon } = locationSelection;

  fetchApi(
    `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${apiKey}&units=metric`
  )
    // .then((res) => res.json())
    .then((data) => console.log(data))
    .catch((error) => {
      console.log('something went wrong');
    });
});
